// Minimal test to reproduce the VerifyError
type Result = Ok(Int) | Error(String)

val result = Ok(42)
val message = match result {
    case Ok(value) => "Success: " + value
    case Error(msg) => "Error: " + msg
}
// Don't use assert for now to isolate the pattern matching issue
println("Done")